packages/adapter-cloudflare/files/worker.js

```diff
@@ -45,41 +45,14 @@ export default {
 
 		// dynamically-generated pages
 		try {
-			const rendered = await app.render({
+			return await app.render({
 				url,
 				rawBody: new Uint8Array(await req.arrayBuffer()),
 				headers: Object.fromEntries(req.headers),
 				method: req.method
 			});
-
-			if (rendered) {
-				return new Response(rendered.body, {
-					status: rendered.status,
-					headers: make_headers(rendered.headers)
-				});
-			}
 		} catch (e) {
 			return new Response('Error rendering route: ' + (e.message || e.toString()), { status: 500 });
 		}
-
-		return new Response({
```

Comment:
i'm curious, why isn't 404 handling needed anymore in `adapter-cloudflare` but remains in `adapter-node`? you may want to weigh in on https://github.com/sveltejs/kit/issues/3363#issuecomment-1013919393 which seems like it's quite related to this change

Reply:
It was a mistake that it was there in the first place — `app.render` will always return a response (unless we make the change #3363 is asking for)